java/arcs/core/data/proto/policy.proto

```diff
@@ -0,0 +1,107 @@
+syntax = "proto3";
+
+package arcs;
+
+option java_package = "arcs.core.data.proto";
+option java_multiple_files = true;
+
+// Defines a data usage policy.
+message PolicyProto {
+  // Policy name.
+  string name = 1;
+
+  // Targets governed by the policy.
+  repeated PolicyTargetProto targets = 2;
+
+  // Additional config metadata for the policy.
+  repeated PolicyConfigProto configs = 3;
```

Comment:
How about making this a map<string, proto.Any>, and using it as a container for any feature-specific thing that Arcs doesn't need to "understand"? This way you could also get rid of fields like egress_type, user_consent_required, and approval_condition

Reply:
I thought about using Any, but the trouble is that something, somewhere needs to define the custom protos and fill them in. The code that fills in this proto (the parser, and then the AST-to-proto converter) are Arcs core code, and those code modules aren't going to be aware of custom client protos. I couldn't see anyway of assembling the proto without moving custom client code directly into either of those code modules (parser or converter).

Reply:
Could we have clients define an Arcs schema and convert it to an Any proto under the hood?

Regardless whether this is a proto or a map<string, string>, it might still be a good thing removing from the internal representation of the policy all fields that Arcs does not need to "understand" (egress_type, user_consent, approval_condition) so that we can keep the policy clean from customer specific stuff. We can then workout the best way to support typed customer configs (Any proto or something else) at a later stage.

Reply:
You need to have a concrete definition for your proto in order to stash it as an Any, you can't just assemble one dynamically. Representing dynamic data is the kind of thing that google.protobuf.Struct is for, which is essentially proto-style JSON encoding. I thought about using it here; however, it's usage is restricted inside Google (they prefer everyone define a dedicated proto for themselves because it's smaller and faster), which is why I ultimately went with map<string, string>. That should get us pretty far, and we can make something more sophisticated later if it's becomes important.

Reply:
I've removed user_consent and approval_condition from the proto, but Arcs needs to "understand" egress_type, since it needs to be enforced by the policy engine.

Reply:
Not sure we're asking you to enforce the egress type but if you want to leave this that's fine. I think you'll end up having to support more specific egress types at some point though, but we can cross that bridge when we get there.